llvm/include/llvm/IR/IRBuilder.h

```diff
@@ -1722,16 +1722,19 @@ class IRBuilderBase {
     return Insert(BinOp, Name);
   }
 
-  Value *CreateLogicalAnd(Value *Cond1, Value *Cond2, const Twine &Name = "") {
+  Value *CreateLogicalAnd(Value *Cond1, Value *Cond2, const Twine &Name = "",
+                          Instruction *MDFrom = nullptr) {
     assert(Cond2->getType()->isIntOrIntVectorTy(1));
     return CreateSelect(Cond1, Cond2,
-                        ConstantInt::getNullValue(Cond2->getType()), Name);
+                        ConstantInt::getNullValue(Cond2->getType()), Name,
+                        MDFrom);
```

Comment:
I *think* it's OK to copy all MD here - presumably callers that don't want any will drop explicitly? I'll defer to @nikic .

Otherwise we could have a instcombine variant that does this.

Separately: the `ProfcheckDisableMetadataFixes` flag?

Reply:
Only prof and unpredicatable metadata get copied from MDFrom.